src/lib.rs

```diff
@@ -870,6 +876,8 @@ impl Format {
             Format::Match(_, branches) => branches.iter().any(|(_, f)| f.depends_on_next(module)),
             Format::Dynamic(_name, _dynformat, f) => f.depends_on_next(module),
             Format::Apply(..) => false,
+            // REVIEW - is this correct?
+            Format::ForEach(_expr, _lbl, f) => f.depends_on_next(module),
```

Comment:
I think this is correct yes, if we permit f to be flexible, which we probably do.